api/v1/storagecluster_types.go

```diff
@@ -94,6 +94,48 @@ type StorageClusterSpec struct {
 	// +optional
 	// +nullable
 	LogCollector *rookCephv1.LogCollectorSpec `json:"logCollector,omitempty"`
+
+	// BackingStorageClasses is a list of storageclasses that will be used for the
+	// storageDeviceSets.
+	BackingStorageClasses []BackingStorageClass `json:"backingStorageClasses,omitempty"`
+	// DefaultStorageProfile is the default storage profile to use for
+	// the storageclassclaims as StorageProfile is optional.
+	DefaultStorageProfile string `json:"defaultStorageProfile,omitempty"`
+
+	StorageProfiles []StorageProfile `json:"storageProfiles,omitempty"`
+}
+
+// StorageProfile is the storage profile to use for the storageclassclaims.
+type StorageProfile struct {
+	// Name of the storage profile.
+	Name string `json:"name,omitempty"`
+	// DeviceClass is the deviceclass name.
+	DeviceClass string `json:"deviceClass,omitempty"`
+	// configurations to use for cephfilesystem.
+	SharedFilesystemConfiguration SharedFilesystemConfigurationSpec `json:"sharedFilesystemConfiguration,omitempty"`
+	// configurations to use for  profile specific blockpool.
+	BlockPoolConfiguration BlockPoolConfigurationSpec `json:"blockPoolConfiguration,omitempty"`
+}
+
+type SharedFilesystemConfigurationSpec struct {
+	Parameters map[string]string `json:"parameters,omitempty"`
+}
+
+type BlockPoolConfigurationSpec struct {
+	Parameters map[string]string `json:"parameters,omitempty"`
+}
+
+// BackingStorageClass defines the backing storageclass for StorageDeviceSet
+type BackingStorageClass struct {
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Provisioner indicates the type of the provisioner.
+	// +optional
+	Provisioner string `json:"provisioner,omitempty"`
+
+	// Parameters holds the parameters for the provisioner that should
+	// create volumes of this storage class.
+	// +optional
+	Parameters map[string]string `json:"parameters,omitempty"`
```

Comment:
Yes, we can do it, IMO it doesn't make sense to reuse storageclass as we will not use all the fields in the storageclass even if the user passes, it can create confusion, please let me know what you think. if required I can check it.

Reply:
Trying to understand your comment. 
If the reason is that we want our backing storage classes to be opinionated then I am fine with this solution. But can we stick to the exact sub struct of the StorageClass substruct (field names, type and serialization semantics)